queries.sql

```diff
@@ -8,3 +8,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE (weight_kg >= 10.4) AND (weight_kg <= 17.3);
+
+/* Transaction */
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'digimon' WHERE name IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT birth;
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
```

Comment:
- [ ] Ouch! this is a close one but the rollback command at line `37` should be to the specific savepoint you made which is `birth` so kindly update this. And the transaction you created starting from line `39` should be a part of the first transaction and not a separate one so please make sure to put this in the first transaction.